src/platform/nrfconnect/DFUSync.cpp

```diff
@@ -0,0 +1,34 @@
+#include "DFUSync.h"
+#include <crypto/RandUtils.h>
+
+CHIP_ERROR DFUSync::Take(uint32_t & id)
+{
+    if (mIsTaken)
+    {
+        if (id == mOwnerId)
+        {
+            return CHIP_NO_ERROR;
+        }
+        return CHIP_ERROR_BUSY;
+    }
+
+    /* Generate random owner id to make this allocation unique. */
+    mIsTaken = true;
+    mOwnerId = chip::Crypto::GetRandU32();
```

Comment:
Have you considered introducing some limits on the number of supported users of this module? By doing so we could get rid of the random generator usage and simply assign some predefined values, or monotonic counter. Asking especially because of the fact that we have only 2 DFUs mechanisms implemented.

Reply:
I didn't want to use predefined values, as this module theoretically could be used for any type of dfu in the future and maybe it should not care about it. Yeah, we could use monotonic counter, I don't know if this will be a big time optimization, but ok

Reply:
Replaced rand with counter.